src/ol/controls/RoutingControl.js

```diff
@@ -215,6 +218,63 @@ class RoutingControl extends Control {
 
     this.loading = true;
 
+    // Clear source
+    this.routingLayer.olLayer.getSource().clear();
+
+    // Create point features for the viaPoints
+    this.viaPoints.forEach((viaPoint, idx) => {
+      return this.drawViaPoint(viaPoint, idx);
+    });
+
+    this.graphs
+      .filter(([graph]) => graph !== 'osm')
+      .forEach(([graph], index) =>
+        this.api
+          .route(
```

Comment:
to avoid duplication of code . Could we track all the promises then return  Promise.all() ?

Reply:
Also really good idea ✅